File: test/com/barrybecker4/common/search/slidingpuzzletests/Watch.java
package com.barrybecker4.common.search.slidingpuzzletests;


public class Watch {

    private final long start;

    public Watch() {
        start = System.currentTimeMillis();
    }

    /**
     * Returns the elapsed time (in seconds) since this object was created.
     */
    public double getElapsedTime() {
        long now = System.currentTimeMillis();
        return (now - start) / 1000.0;
    }

}

File: scala-test/com/barrybecker4/common/Watch.scala
/* Copyright by Barry G. Becker, 2000-2018. Licensed under MIT License: http://www.opensource.org/licenses/MIT */
package com.barrybecker4.common

class Watch {
  final private val start = System.currentTimeMillis

  /** @return the elapsed time (in seconds) since this object was created. */
  def getElapsedTime: Double = (System.currentTimeMillis - start) / 1000.0
}
